src/Bicep.Cli/Commands/BuildParamsCommand.cs

```diff
@@ -54,10 +67,49 @@ public async Task<int> RunAsync(BuildParamsArguments args)
                 return 1;
             }
 
+            var paramsOverrides = Environment.GetEnvironmentVariable("BICEP_PARAMETERS_OVERRIDES");
+
+            if (paramsOverrides is { })
+            {
+                var paramsOverridesJson = JObject.Parse(paramsOverrides);
+
+                var fileContents = await File.ReadAllTextAsync(paramsInputPath);
+                var sourceFile = SourceFileFactory.CreateBicepParamFile(PathHelper.FilePathToFileUrl(paramsInputPath), fileContents);
+
+                var newProgramSyntax = CallbackRewriter.Rewrite(sourceFile.ProgramSyntax, syntax =>
```

Comment:
>Rewrite [](http://example.com/codeflow?start=56&length=7)

Since we're modifying the user-provided file, what happens to the locations in the TextSpan objects on the generated syntax nodes? If the compiler decides to emit a diagnostic whose span involves one of the generated nodes, are we going to emit misleading locations?

This is likely out of scope for this PR, but I'm wondering if we need to add a concept of an "external" location for syntax nodes for scenarios such as this. (Plus semantics on how an external location interacts with regular locations when we do TextSpan calculations in the compiler.)

Reply:
The issue with line numbers is something we've discussed during parameters sync meeting multiple times. The solution two for question one in [design gist](https://gist.github.com/Usman0111/92dda10627e550737341c74d617f4fa6#1-how-would-inline-params-would-be-validated) somewhat attempts to solve this issue. However, we've decided to continue with current implementation as it is lower cost and errors are "good enough" for users to debug.

Adding an "external" location sounds like a great idea. I've linked this comment to the gist so we can track this in future discussions